### ISSUE TITLE
Problem with "FindOne" and "FindAll" functions, Invoke-ADCheckAll is not working.

### ISSUE THREAD

**Issue by EscrVirtuoso** [0xJs/domain_audit#5]
**Describe the bug**
I have some problems with FindOne and FindAll functions even if I followed every installation step properly. I'm not sure if this is a bug or just a little problem, but I'm asking some help here, maybe some people have the same problem.

**To Reproduce**
I just installed everything in the right order, and launched the ADCheckAll command with the good options. 
Right after this, I had all the errors I put in screenshot. I can't have any informations about my Active Directory because of these errors.

**Expected behavior**
I just expected the script to work as I installed everything, and checked multiple times if everything was ok. But this isn't working unfortunately.

Thank you very much for reading this ! And thank you even more if you have some informations to help me.

**Screenshots**
![image](https://github.com/0xJs/domain_audit/assets/168753295/c430b0db-2174-4aab-9c26-54304fc48823)

**Comment by 0xJs** [comment]
Hey,If you run Powerview commands manually with a credential object, the credentials, domains and server parameters with the values it works?Those errors are from the powerview commands because it can't reach or interact with the DC.Sent from Outlook for Android

**Comment by 0xJs** [comment]
And did you run it as Admin? Might be a DNS issue.Sent from Outlook for Android

**Comment by EscrVirtuoso** [comment]
Thanks for your response !

I'm running it as Admin for sure.

I will try to run Powerview commands apart, but I'm not sure of how I can do this, I am kinda new into powershell commands and things like that.

I will put some other screenshots after if it can help !

**Comment by 0xJs** [comment]
Hi EscrViortuso,

 

Try to run a simple querry to target the users. For example

 

 

Load the PowerView Module first from the import directory of my tool:

 

. ./PowerView.ps1

 

Create a credential object with:

 

$Creds = Get-Credential

 

Then run a query:

 

Get-DomainUser -Domain <DOMAIN NAME>  -Server <DC IP> -Credential $Creds

 

If that gives any error there probably is a problem with the credentials or
connectivity to the domain.

**Comment by EscrVirtuoso** [comment]
Thanks ! I tried what you said, here is a screenshot of what I have as an error and what I did to have this error.

I also reinstalled everything properly, I don't really know what the problem is actually.

Thank you very much anyway, that's very cool to help me :)

![image](https://github.com/0xJs/domain_audit/assets/168753295/8350d0cd-4e7e-4ad3-825a-c072005fa2da)

**Comment by 0xJs** [comment]
Hi,

 

No problem to help! I think there are connectivity issues. Are you running
this in a lab?

 

Can you do a nmap scan of the target host and check if LDAP (port 389) is
available.

**Comment by EscrVirtuoso** [comment]
I found the problem ! Thank you very much for your help.

I had a problem with the name of the Active Directory domain, that's why it was not working !

I still have some errors at the end of the check scan, I will send you screenshots when I can, but that's maybe just some errors with no impact, as the script is now working well !

Thank you very much for your work and your help !

**Comment by EscrVirtuoso** [comment]
Hello again ! I'm so sorry to send you a message !!

I have errors during the scan with a module named imp and cme with crackmapexec.

Here are my errors : 

![image](https://github.com/0xJs/domain_audit/assets/168753295/aee041a1-b2a3-4468-be71-6286a7fd23ae)


Do you have an idea on how to resolve this ?

Thank you very much for your work !

**Comment by 0xJs** [comment]
Hey,Probably because Windows Defender deleted some files from crackmapexec. Completely disable defender forever through local group policies or allow list the user directory with the .shiv directory.Then reinstall and get crackmapexec working individually before running domain audit. You can test the commandPython pathtocme smb IP -u user -p password On my phone atm so can't be more specificSent from Outlook for AndroidFrom: EscrVirtuoso ***@***.***>Sent: Tuesday, May 28, 2024 3:00:19 pmTo: 0xJs/domain_audit ***@***.***>Cc: 0xJs ***@***.***>; Comment ***@***.***>Subject: Re: [0xJs/domain_audit] Problem with "FindOne" and "FindAll" functions, Invoke-ADCheckAll is not working. (Issue #5)
Hello again ! I'm so sorry to send you a message !!
I have errors during the scan with a module named imp and cme with crackmapexec.
Here are my errors :
image.png (view on web)
Do you have an idea on how to resolve this ?
Thank you very much for your work !

**Comment by EscrVirtuoso** [comment]
Thanks for your response ! 

I tried to disable windows defender but I have the same error for the moment. 

When I'm using crackmapexec only it seems to work, here is a screenshot : 

![image](https://github.com/0xJs/domain_audit/assets/168753295/34357d54-1429-431a-8293-c84c02af6a2a)

I will try to see what this .shiv directory is and how can I change it to make the tool work. 

Have a nice day and thank you very much :)

**Comment by 0xJs** [comment]
Yeah cme is working fine it seems. You still get the same error when running the domain audit tool?It is complaining about packages. Remove the directory and rerun crackmapexec it should recreate that .shiv directory and files.Sent from Outlook for AndroidFrom: EscrVirtuoso ***@***.***>Sent: Wednesday, May 29, 2024 5:25:43 pmTo: 0xJs/domain_audit ***@***.***>Cc: 0xJs ***@***.***>; Comment ***@***.***>Subject: Re: [0xJs/domain_audit] Problem with "FindOne" and "FindAll" functions, Invoke-ADCheckAll is not working. (Issue #5)
Thanks for your response !
I tried to disable windows defender but I have the same error for the moment.
When I'm using crackmapexec only it seems to work, here is a screenshot :
image.png (view on web)
I will try to see what this .shiv directory is and how can I change it to make the tool work.
Have a nice day and thank you very much :)

**Comment by EscrVirtuoso** [comment]
Well, even when I try to remove the .shiv directory and then run the cme alone there's no problem, but once again when using the tool I have the same error. I don't really understand what is the problem, it is kinda strange. 

I don't know if it is normal too, but I have a directory .cme which contains some other directories which seems empty (for example the folder module)

![image](https://github.com/0xJs/domain_audit/assets/168753295/516308bd-99dd-4f1d-9123-c9e911f1a7ba)

And here is my .shiv folder : 

![image](https://github.com/0xJs/domain_audit/assets/168753295/436c7c35-da18-4ba6-b53f-f3993e24f314)

With a lot of other folders in the site-packages like this : 

![image](https://github.com/0xJs/domain_audit/assets/168753295/f047d169-c6b8-491c-9d6b-b8f4028f43cc)

Does everything seems normal and good ?

**Comment by 0xJs** [comment]
Sorry no clue either from looking at the screenshots :( troubleshooting is a bitch. Will try it out next week if it still works for me.You can try to look up the command it's trying to run and run it manually. Can't really see it in the screenshot. But you can open the script and search for the module that is running crackmapexec.Sent from Outlook for AndroidFrom: EscrVirtuoso ***@***.***>Sent: Wednesday, May 29, 2024 6:46:36 pmTo: 0xJs/domain_audit ***@***.***>Cc: 0xJs ***@***.***>; Comment ***@***.***>Subject: Re: [0xJs/domain_audit] Problem with "FindOne" and "FindAll" functions, Invoke-ADCheckAll is not working. (Issue #5)
Well, even when I try to remove the .shiv directory and then run the cme alone there's no problem, but once again when using the tool I have the same error. I don't really understand what is the problem, it is kinda strange.
I don't know if it is normal too, but I have a directory .cme which contains some other directories which seems empty (for example the folder module)
image.png (view on web)
And here is my .shiv folder :
image.png (view on web)
With a lot of other folders in the site-packages like this :
image.png (view on web)
Does everything seems normal and good ?

**Comment by EscrVirtuoso** [comment]
Hello again ! 
I find a solution to the problem when I checked all my files once again. I had a problem with some python versions which were not the good ones and those versions were used by your tool, it caused a lot of problems :/ 

Well, now it seems to work, but I have just one more question. The step "Checking for access over RDP" isn't completing, it seems to be locked or bugged I don't know, but maybe it is just taking a long time to complete ?

Here is a screenshot :

![image](https://github.com/0xJs/domain_audit/assets/168753295/b6d987bf-852b-4ec5-8851-b9d9d7fa99d6)

I really want to thank you for your help, I learned a lot of things and it's very nice of you :D

**Comment by 0xJs** [comment]
Glad that you were able to fix it!

 

Yeah the CME part is hanging a lot, you might need to enter or to wait. It
is probing every system through port 3389 with cme. Cme rdp <IP> -u <USER>
-p <PASSWORD>.

 

Should probably replace it with netexec (which my friend did and made a pull
request) but haven't been able to test it in my lab yet. Busy with other
things.
